Pendual: Finish the majority of the events

trmazi opened the pull request:
In this commit I have pushed my current changes to IIDX 22 Pendual. I've completed every event other than Qpronicle Chord as I have ran out of steam on this mix for now. I will return and try to 100% this game sooner than later though. For now i'll leave it at this.

I have made the events all Frontend modifiable so that's good enough for now. I've thrown in a ton of #TODO statements to show my current progress.

trmazi committed:
Pendual: Finish the majority of the events

In this commit I have pushed my current changes to IIDX 22 Pendual. I've completed every event other than Qpronicle Chord as I have ran out of steam on this mix for now. I will return and try to 100% this game sooner than later though. For now i'll leave it at this.

I have made the events all Frontend modifiable so that's good enough for now. I've thrown in a ton of #TODO statements to show my current progress.

DragonMinded commented:
This is kind-of a mess but I'll try to review it anyway. General comments: why are you submitting this with a bunch of TODOs added when it is adding events for a single mix? Surely you could button it up before sending it in. As it stands its not mergeable.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -134,10 +134,10 @@ def get_settings(cls) -> Dict[str, Any]:
                     'setting': 'global_shop_ranking',
                 },
                 {
-                    'name': 'Events In Omnimix',
```
Why did you get rid of this switch? Every other IIDX supports disabling events in omnimix.

trmazi replied:
I removed it and had forgotten to re-implement it. Will be fixed in the next commit.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -153,6 +153,18 @@ def get_settings(cls) -> Dict[str, Any]:
+                    'values': {
+                        0: 'No Event',
+                        1: 'Chrono Seeker',
+                        #2: 'Qpronicle Chord', TODO: Finish this event
```
What's remaining?

trmazi replied:
I need to figure out the achievements section of the event. For some reason the game kills off the event if I send through it after saving in unformat_profile.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -889,46 +900,40 @@ def handle_IIDX22pc_request(self, request: Node) -> Optional[Node]:
             limit = Node.void('limit')
             root.add_child(limit)
-            limit.set_attribute('phase', '24')
+            limit.set_attribute('phase', '21')
```
Is this the real max?

trmazi replied:
from my understanding, yes.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -889,46 +900,40 @@ def handle_IIDX22pc_request(self, request: Node) -> Optional[Node]:
                 timeshift_override = 0
-                omni_events = False
-
-            if self.omnimix and (not omni_events):
```
Second comment on this, why remove this?

trmazi replied:
See explanation above in the first comment.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -889,46 +900,40 @@ def handle_IIDX22pc_request(self, request: Node) -> Optional[Node]:
             common_cd_event = Node.void('common_cd_event')
             root.add_child(common_cd_event)
-            common_cd_event.set_attribute('open_list', '3')
+            common_cd_event.set_attribute('open_list', str(3))
```
Why this change?

trmazi replied:
I had messed around with changing the values with frontend, meant to set it back to just being a string.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -1185,6 +1190,35 @@ def handle_IIDX22pc_request(self, request: Node) -> Optional[Node]:
+            cm.set_attribute('folder', 'cm_event3')
+            cm.set_attribute('id', "0")
+
+            #Ranking details in title screen
```
Nice.

trmazi replied:
these are separate: comment goes to the rankings under, though this is not actually what they're for or im missing part of the puzzle. was meant to be looked at again, had not gotten to that.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -1185,6 +1190,35 @@ def handle_IIDX22pc_request(self, request: Node) -> Optional[Node]:
+                rankingcharts.append(mid)
+            root.add_child(Node.u16_array('monthly_mranking', rankingcharts))
+            root.add_child(Node.u16_array('total_mranking', rankingcharts))
+            ##need to add in the top 20 rankings for month and total network
```
Is that not what the above section does? Except for it doesn't really do it for the last month.

trmazi replied:
need to fix this with the proper values

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -1707,6 +1741,172 @@ def format_profile(self, userid: UserID, profile: Profile) -> Node:
+        cxb_music = Node.void('cc_collabo_music')
+        root.add_child(cxb_music)
+        cxb_music.set_attribute('music_list', "1")
+        #TODO: Find the proper music_list number
```
Seems this isn't finished? Did you check bemaniwiki?

trmazi replied:
I did not think that there would be any information there about the coke event music details. I plan on finishing that properly in the next commit.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -1707,6 +1741,172 @@ def format_profile(self, userid: UserID, profile: Profile) -> Node:
+        #sandglass event details
+        time_sandglass = Node.void('time_sandglass')
+        root.add_child(time_sandglass)
+        time_sandglass.set_attribute('item_num', "99")
```
So this isn't done at all? Is it going to be done like infinite paseli checks?

trmazi replied:
Yes, this is set to infinite. Konami added sandglasses to your profile by buying them with paseli and the game removes them from your account at logout. I could make these purchasable on frontend similar to paseli, but for now they're set to infinite.

DragonMinded commented on bemani/backend/iidx/pendual.py:
```diff
@@ -1707,6 +1741,172 @@ def format_profile(self, userid: UserID, profile: Profile) -> Node:
+        pyramid.set_attribute('statue_0', str(0))
+        pyramid.set_attribute('statue_1', str(0))
+        pyramid.set_attribute('statue_2', str(0))
+        #TODO: Figure out how to actually send this data. 
```
I gave up reviewing here. This seems like a bunch of hardcoded stuff to make the event *show up* but it doesn't actually implement any of the logic to make the event *work*.

trmazi replied:
This does not cause the pyramid event to show up. i suppose i should have removed them from my PR as they're basically useless as they stand and are not required. From what i can tell, this event will only trigger on an older build of pendual, so i might as well just not bother with it at all.

DragonMinded reviewed (changes requested):
I don't think you ran lint or type checking? There's stuff I can see at a glance would set that off.

trmazi committed:
Pendual: Remove clutter and add back omni-event control

In this commit I have re-added the option to disable events in omnimix and have remove code that was not needed/not fully finished that I added in the past commit. I've also reverted changes that were not relevant nor useful.

Subject38 commented:
Having looked over at the changes, everything looks fine to me, honestly confused why there hasn't been further review...
